data/basics/testing/roundtrip/roundtrip.go

```diff
@@ -0,0 +1,197 @@
+	randomOpts     []protocol.RandomizeObjectOption
+	rapidGen       interface{} // *rapid.Generator[A], stored as interface{} to avoid type parameters
+	useRapid       bool
+	skipNearZeros  bool
+}
+
+type randomCountOption int
+
+func (n randomCountOption) apply(cfg *checkConfig) {
+	count := int(n)
+	cfg.randomCount = &count
+}
+
+type randomOptsOption []protocol.RandomizeObjectOption
+
+func (opts randomOptsOption) apply(cfg *checkConfig) {
+	cfg.randomOpts = append(cfg.randomOpts, opts...)
+}
+
+type rapidGenOption struct {
+	gen interface{}
+}
+
+func (r rapidGenOption) apply(cfg *checkConfig) {
+	cfg.rapidGen = r.gen
+	cfg.useRapid = true
+}
+
+// Opts configures round-trip checking behavior.
+// The first argument specifies the number of random test cases to generate.
+// Additional protocol.RandomizeObjectOption arguments can be passed to customize randomization.
+func Opts(count int, opts ...protocol.RandomizeObjectOption) CheckOption {
+	return multiOption{randomCountOption(count), randomOptsOption(opts)}
+}
+
+// NoRandomCases disables RandomizeObject testing (but still runs NearZeros).
+// Use this when RandomizeObject generates invalid values for constrained types.
+// Combine with NoNearZeros() to disable all automatic testing.
+func NoRandomCases() CheckOption {
+	return randomCountOption(0)
+}
+
+type skipNearZerosOption struct{}
+
+func (skipNearZerosOption) apply(cfg *checkConfig) {
+	cfg.skipNearZeros = true
+}
+
+// NoNearZeros disables NearZeros testing, only using RandomizeObject for random variants.
+// Use this for non-struct types (maps, slices) where NearZeros doesn't apply.
+func NoNearZeros() CheckOption {
+	return skipNearZerosOption{}
+}
+
+// WithRapid specifies a rapid.Generator to use for property-based testing.
+// If provided, rapid.Check will be used instead of protocol.RandomizeObject (runs 100 tests).
+func WithRapid[A any](gen *rapid.Generator[A]) CheckOption {
+	return rapidGenOption{gen: gen}
+}
+
+type multiOption []CheckOption
+
+func (m multiOption) apply(cfg *checkConfig) {
+	for _, opt := range m {
+		opt.apply(cfg)
+	}
+}
+
+// Check verifies that converting from A -> B -> A yields the original value.
+// By default, tests the provided example, all NearZeros variants (one per field),
+// and 100 randomly generated values using protocol.RandomizeObject.
+// Use WithRapid to provide a custom rapid.Generator for property-based testing.
+// Use NoRandomCases to disable RandomizeObject (still runs NearZeros).
+// Use NoNearZeros to disable NearZeros (for non-struct types like maps).
+// Use Opts to customize the number of random tests or pass RandomizeObjectOptions.
+func Check[A any, B any](t *testing.T, a A, toB func(A) B, toA func(B) A, opts ...CheckOption) bool {
+	cfg := checkConfig{}
+	for _, opt := range opts {
+		opt.apply(&cfg)
+	}
+
+	// Test the provided example first
+	if !checkOne(t, a, toB, toA) {
+		t.Errorf("Round-trip failed for provided example: %+v", a)
+		return false
```

Comment:
Why Errorf and report passing as a boolean?  I would think things that take a testing.T handle failure internally.  There's no advantage to calling `require.True` around `Check` is there?